packages/jupyter-ai/jupyter_ai/actors/learn.py

```diff
@@ -83,26 +110,60 @@ def _process_message(self, message: HumanChatMessage):
             default_splitter=RecursiveCharacterTextSplitter(chunk_size=self.chunk_size, chunk_overlap=self.chunk_overlap)
         )
 
-        loader = RayRecursiveDirectoryLoader(load_path)
-        texts = loader.load_and_split(text_splitter=splitter)        
+        loader = RayRecursiveDirectoryLoader(path)
+        texts = loader.load_and_split(text_splitter=splitter)
         self.index.add_documents(texts)
-        self.save()
-
-        response = f"""🎉 I have indexed documents at **{load_path}** and I am ready to answer questions about them. 
-        You can ask questions about these docs by prefixing your message with **/ask**."""
-        self.reply(response, message)
-
-    def get_index(self):
-        return self.index
+        self.add_dir_to_metadata(path)
+    
+    def add_dir_to_metadata(self, path: str):
+        dirs = self.metadata.dirs
+        index = next((i for i, dir in enumerate(dirs) if dir.path == path), None)
+        meta = IndexedDir(path=path)
+        if index:
+            dirs[index] = meta
+        else:
+            dirs.append(meta)
+        self.metadata.dirs = dirs
+
+    def delete_and_relearn(self):
+        if not self.metadata.dirs:
+            self.delete()
+            return
+        message = """🔔 Hi there, It seems like you have updated the embeddings model. For the **/ask**
+        command to work with the new model, I have to re-learn the documents you had previously 
+        submitted for learning. Please wait to use the **/ask** command until I am done with this task."""
+        self.reply(message)
+        
+        metadata = self.metadata
+        self.delete()
+        self.relearn(metadata)
 
     def delete(self):
         self.index = None
+        self.metadata = IndexMetadata(dirs=[])
         paths = [os.path.join(self.index_save_dir, self.index_name+ext) for ext in ['.pkl', '.faiss']]
         for path in paths:
             if os.path.isfile(path):
                 os.remove(path)
         self.create()
 
+    def relearn(self, metadata: IndexMetadata):
+        # Index all dirs in the metadata
+        if not metadata.dirs:
+            return    
+        dirs = []
+        for dir in metadata.dirs:
+            self.learn_dir(dir.path)
+            dirs.append(dir.path)
```

Comment:
The temporary `dirs` variable stores the list of indexed directories for the reply message. However, this can be substituted with `self.metadata.dirs` when building the reply message, meaning that there's no reason to define and maintain this temporary variable.

Reply:
The `dirs` variable simplifies the storage of just the `paths`, which are accessed while building the list for the reply message. The other option is to omit the `dirs` variable, and use comprehension to build the `dir_list` string.